File: src/main/java/com/chylb/model/effort/Effort.java
package com.chylb.model.effort;

import com.chylb.model.distance.Distance;
import com.chylb.model.activity.Activity;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

public class Effort {
    @Id
    @GeneratedValue
    private long id;

    @ManyToOne
    private Activity activity;

    @ManyToOne
    private Distance distance;

    private int time;
}

File: frontend/src/types/effort.ts
import { Activity } from "./activity";
import { Distance } from "./distance";

export interface Effort {
    id: number;
    activity: Activity;
    distance: Distance;
    time: number;
    ordinal: number;
    rank: number;
    ix0: number;
    ix1: number;
    flagged: boolean;
}